scripts/save_y2logs

```diff
@@ -7,110 +7,140 @@
 	;;
 
     *)
-	echo "FATAL: Uncompressed archives not supported" >&2
-	echo "Use one of: .tgz .tar.gz .tar.bz2"          >&2
+	echo "FATAL: argument is not a (compressed) tar file." >&2
+	echo "Use one of these suffixes: .tar .tgz .tar.gz .tar.xz .tar.bz2" >&2
 	exit 4
   esac
 fi
 
+# check for compression program
+if [ "$COMPRESSION" -a ! -x "/usr/bin/$COMPRESSION" ]; then
+  echo "FATAL: /usr/bin/$COMPRESSION not available" >&2
+  exit 3
+fi
+
 TMPDIR=`mktemp -d` || ( echo "FATAL: Failed to create a temporary directory" >&2; exit 5 );
-LIST=YaST2
 
-if [ -f /var/log/evms-engine.log ]; then
-    LIST="$LIST $( cd /var/log/; ls evms-engine.*)"
-fi
 
-if [ -x /bin/dmesg ]; then
-    dmesg > $TMPDIR/dmesg
-    LIST="$LIST $TMPDIR/dmesg"
-fi
+# 1. gather some information in $TMPDIR
 
-if [ -f /var/log/zypp/history ]; then
-    LIST="$LIST $( cd /var/log/; ls zypp/history*)"
+# last kernel messages, if any
+if [ -x /bin/dmesg ]; then
+  dmesg > $TMPDIR/dmesg
+  LIST_TMP="$LIST_TMP dmesg"
 fi
 
+# strip sensitive information
 if [ -f /etc/install.inf ]; then
-    sed 's/\(^WlanESSID: \|^WlanKey: \|^RootPassword: \|^VNCPassword: \|^Password: \).*$/\1XXXXXX/' </etc/install.inf >$TMPDIR/install.inf
-    LIST="$LIST $TMPDIR/install.inf"
+  sed 's/\(^WlanESSID: \|^WlanKey: \|^RootPassword: \|^VNCPassword: \|^Password: \).*$/\1XXXXXX/' </etc/install.inf >$TMPDIR/install.inf
+  LIST_TMP="$LIST_TMP install.inf"
 fi
 
-if [ -f /etc/X11/xorg.conf ]; then
-    LIST="$LIST /etc/X11/xorg.conf"
+# if storing logs at the end of installation after bootloader fail, try to store pbl log from target system
+if [ -f /mnt/var/log/pbl.log ]; then
+  cp /mnt/var/log/pbl.log $TMPDIR/pbl-target.log
+  LIST_TMP="$LIST_TMP pbl-target.log"
 fi
 
-if [ -f /etc/X11/xorg.conf.install ]; then
-    LIST="$LIST /etc/X11/xorg.conf.install"
+# installed packages
+if [ -f /var/lib/rpm/Packages ]; then
+  rpm -qa --qf '%{NAME}-%{VERSION}-%{RELEASE}\t\t\t(%{VENDOR})\t%{DISTRIBUTION}\n' 2>/dev/null | sort >/$TMPDIR/rpm-qa
+  LIST_TMP="$LIST_TMP rpm-qa"
 fi
 
-if [ -f /var/log/Xorg.0.log ]; then
-    LIST="$LIST /var/log/Xorg.0.log"
+# rename, so people can see it
+if [ -f /.packages.root ]; then
+  cp /.packages.root $TMPDIR/_packages.root
+  LIST_TMP="$LIST_TMP _packages.root"
 fi
 
-if [ -f /var/log/zypper.log ]; then
-    LIST="$LIST /var/log/zypper.log"
-fi
 
-if [ -f /var/log/pk_backend_zypp ]; then
-    LIST="$LIST /var/log/pk_backend_zypp"
-fi
+# 2. things from /var/log relevant for us
 
-if [ -f /var/log/pbl.log ]; then
-    LIST="$LIST /var/log/pbl.log"
-fi
-# if storing logs at the end of installation after bootloader fail, try to store pbl log from target system
-if [ -f /mnt/var/log/pbl.log ]; then
-    LIST="$LIST /mnt/var/log/pbl.log"
-fi
+VAR_LOG_FILES='\
+  YaST2 \
+  Xorg.0.log \
+  zypper.log zypp/history* pk_backend_zypp \
+  pbl.log linuxrc.log wickedd.log \
+  evms-engine.* \
+'
+cd /var/log
+for i in $VAR_LOG_FILES ; do
+  [ -e "$i" ] && LIST_VAR="$LIST_VAR $i"
```

Comment:
ah, dark bash magic, with having it in `''` and then expanded for cycle. OK, I trust to verify it.